subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/UpgradeDependencyPlugin.java

```diff
@@ -169,7 +169,10 @@ public void execute(String openPullRequestBranch) {
             public void execute(final ReplaceVersionTask task) {
                 task.setDescription("Replaces dependency version in build file.");
                 task.mustRunAfter(CHECKOUT_VERSION_BRANCH);
-                task.setVersionUpgrade(upgradeDependencyExtension);
+                task.setDependencyGroup(upgradeDependencyExtension.getDependencyGroup());
```

Comment:
Do we need "deferred configuration" here?

Reply:
I don't think so, dependencyGroup is set basing on project property that is passed with command line invocation. I can't think of another place where it would make sense to set it. What do you think?

Reply:
Maybe it shouldn't be a part of UpgradeDependencyExtension, would be better to create an separate object for fields that are extracted from "dependency" property.

Reply:
My goal was to consider the configuration order when making this change. If you believe that we don't need to use deferred configuration, I trust you :)

upgradeDependencyExtension will be a public object and users can configure it in the build file. Hence, using deferred configuration is desired even if our default use case is "project property". Otherwise the user can set the values on the extension but they are not used. Thoughts?